packages/block-library/src/block/edit-panel/use-previous.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+// Based on https://usehooks.com/usePrevious/.
+export default function usePrevious( value ) {
```

Comment:
Is this useful enough to move to `@wordpress/compose`? If so, I can create a separate PR for that.

Reply:
it would be good to have it under `@wordpress/compose`

Reply:
Alright, I've created a PR to add it to `@wordpress/compose`: #22540.

Reply:
#22540 has been merged, so I've rebased this PR to use `usePrevious` from `@wordpress/compose`.